### PR TITLE
Overhaul Paratroopers Lua API

### PR DESCRIPTION
**Pull request by pchote**
This pr deprecates `SendParatroopers` and `SendParatroopersFrom`, replacing both with a new `ActivateParatroopers` (named for consistency with the Ion Cannon and Nuke APIs) method.

The long deprecated `*Upgrade` methods are also removed.

Ideally we would merge this in time for the playtest to maximize the amount of time that mappers have to adjust their scripts and minimize the amount of time that we have to live with the legacy code - but I am not optimistic this is feasible, considering how long it takes every mission related PR to be reviewed.

I have tested some but nowhere near all of the Lua changes.

### CODE REVIEW

File: OpenRA.Mods.Common/Scripting/Properties/ParatroopersProperties.cs [hunk 1]
```diff
@@ -27,19 +27,30 @@ public ParatroopersProperties(ScriptContext context, Actor self)
...
 		}
 
+		[Desc("Activate the actor's Paratroopers Power. Returns the aircraft that will drop the reinforcements.")]
+		public Actor[] ActivateParatroopers(WPos target, int facing = -1)
+		{
+			var actors = pp.SendParatroopers(Self, target, facing);
+			return actors.First;
+		}
+
 		[Desc("Activate the actor's Paratroopers Power. Returns the dropped units.")]
```
Comment (teinarss) [inline OpenRA.Mods.Common/Scripting/Properties/ParatroopersProperties.cs]: For consistency we could add DEPRECATED! Will be removed. to the desc.

File: OpenRA.Mods.Common/Scripting/Properties/ParatroopersProperties.cs [hunk 2]
```diff
@@ -27,19 +27,30 @@ public ParatroopersProperties(ScriptContext context, Actor self)
...
 		public Actor[] SendParatroopers(WPos target, bool randomize = true, int facing = 0)
 		{
-			return pp.SendParatroopers(Self, target, randomize, facing);
+			Game.Debug("SendParatroopers is deprecated. Use ActivateParatroopers instead.");
+			var actors = pp.SendParatroopers(Self, target, randomize ? -1 : facing);
+			return actors.Second;
 		}
 
 		[Desc("Activate the actor's Paratroopers Power. Returns the dropped units.")]
```
Comment (teinarss) [inline OpenRA.Mods.Common/Scripting/Properties/ParatroopersProperties.cs]: Same here

**Comment by pchote** [comment]
Fixed and rebased.

File: mods/ra/maps/soviet-08a/soviet08a.lua [hunk 3]
```diff
@@ -53,7 +53,15 @@ end
 AddParadropReinforcementTrigger = function()
 	Trigger.AfterDelay(DateTime.Seconds(90), function()
 		Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
-		scripteddrop.SendParatroopers(ScriptedParadrop.CenterPosition, false, 10)
+		scripteddrop.ActivateParatroopers(ScriptedParadrop.CenterPosition, 10)
+	end)
+end
+
+IdleHunt = function(unit)
```
Comment (teinarss) [inline mods/ra/maps/soviet-08a/soviet08a.lua]: Cant see any changes using this function?

Comment (pchote) [reply mods/ra/maps/soviet-08a/soviet08a.lua]: I think this was a bug fix, but it was now too long ago for me to remember if I tested it.
The `ChurchAmbushTrigger` below calls` IdleHunt` and I guess it wasn't defined before now.

**Review by teinarss — DISMISSED** [review]

File: mods/ra/maps/soviet-08a/soviet08a.lua (hunk 3 above)
Comment (abcdefg30) [inline mods/ra/maps/soviet-08a/soviet08a.lua]: This is already defined in the AI script. Please don't define it again here.

**Comment by pchote** [comment]
Rebased (first push) and fixed (second).

**Review by teinarss — APPROVED** [review]

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/692121cb03ce1849df57c2fbfea3ae5cbb6bfdc9...e00d3165301fa87f6386850ade1c8ef15b4de8b0)

**Pull request merged**